query/data_query.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022-2023 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/clarify/clarify-go/fields"
+)
+
+type dataQuery struct {
+	Filter         dataFilter `json:"filter"`
+	Rollup         string     `json:"rollup,omitempty"`
+	Last           int        `json:"last,omitempty"`
+	Origin         string     `json:"origin,omitempty"`
+	FirstDayOfWeek int        `json:"firstDayOfWeek,omitempty"`
+	TimeZone       string     `json:"timeZone,omitempty"`
+}
+
+// Data holds a data query. Although it does not expose any fields, the type
+// can be decoded from and encoded to JSON.
+type Data struct {
+	query dataQuery
+}
+
+// DataWhere returns a new Data query that joins passed in filters with logical
+// AND.
+func DataWhere(filters ...DataFilter) Data {
```

Comment:
Can we instead add a `DataQuery` function that returns new `Data` but without filters and add a `Filter` method? This will make code more readable from the user's perspective, so the difference between a data query and a resource query will be felt better.
E.g

Reply:
Additionally, regarding naming. To match API namings, we need to call them `DataParams` and `QueryParams` instead of `Data` and `Query`. What do you think?

Reply:
Or even package `query` is more than query, but `params`

Reply:
This has been reworked.